### PR TITLE
Lua appearance listener

### PR DESCRIPTION
**Pull request by maxim-productengine**
Provide abilities to add/replace a specific outfit and wear/remove an individual item.

Demo script shows the list of Outfits (which you can add to COF or replace COF), allows to show the list of selected outfit wearables (and wear or remove selected item).

### CODE REVIEW

File: indra/newview/llappearancelistener.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    LLInventoryModel::item_array_t item_array;
+
+    LLIsType is_category(LLAssetType::AT_CATEGORY);
+    gInventory.collectDescendentsIf(outfits_id, cat_array, item_array, LLInventoryModel::EXCLUDE_TRASH, is_category);
+
+    LLSD outfits_data;
+    for (const LLPointer<LLViewerInventoryCategory> &cat : cat_array)
+    {
+        outfits_data[cat->getUUID().asString()] = cat->getName();
+    }
```
Comment (nat-goodspeed) [inline indra/newview/llappearancelistener.cpp:134]: You inspired me. I looked at these loops and thought: this is a pattern I've seen over and over again, how about [a utility function](https://github.com/secondlife/viewer/pull/1924)?

I stipulate that whether it's clearer in this case is a matter of style. Feel free to tell me that the functions are interesting but the explicit loop makes more sense.

File: indra/newview/llappearancelistener.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+void LLAppearanceListener::wearItem(LLSD const &data)
+{
+    LLAppearanceMgr::instance().wearItemOnAvatar(data["item_id"].asUUID(), true, data["replace"].asBoolean());
+}
+
+void LLAppearanceListener::detachItem(LLSD const &data)
+{
+    LLAppearanceMgr::instance().removeItemFromAvatar(data["item_id"].asUUID());
+}
```
[the remark this answers is not in the document]
Comment (maxim-productengine) [reply indra/newview/llappearancelistener.cpp:124]: Sure, it makes sense.

File: indra/newview/llappearancelistener.cpp [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+void LLAppearanceListener::wearOutfitByName(LLSD const &data)
+{
+    Response response(LLSD(), data);
+    std::string error_msg;
+    if (!LLAppearanceMgr::instance().wearOutfitByName(data["folder_name"].asString(), error_msg, data["append"].asBoolean()))
+    {
+        response.error(error_msg);
+    }
+}
```
[the remark this answers is not in the document]
Comment (maxim-productengine) [reply indra/newview/llappearancelistener.cpp:90]: Hm, I think we may use it for Gallery view in Inventory (where just folder uuid is known). So moving it to `LLAppearanceMgr`.

File: indra/newview/llappearancelistener.cpp (hunk 1 above)
Comment (maxim-productengine) [reply indra/newview/llappearancelistener.cpp]: Actually, it looks good in `getOutfitItems` where there are multiple info fields in the response.

**Review by nat-goodspeed — APPROVED** [review]

**Pull request merged**